btree/btree.c

```diff
@@ -3132,52 +3191,135 @@ static void ff_done(struct slot *slot, bool modified)
 	void            *val_addr;
 	uint64_t         calculated_csum;
 
-	if (modified && h->ff_level == 0 &&
+	if (modified && h->ff_var.ff_level == 0 &&
 	    ff_crctype_get(node) == M0_BCT_BTREE_ENC_RAW_HASH) {
 		val_addr = ff_val(slot->s_node, slot->s_idx);
 		calculated_csum = m0_hash_fnc_fnv1(val_addr, h->ff_vsize);
 		*(uint64_t*)(val_addr + h->ff_vsize) = calculated_csum;
 	}
 }
 
+static void ff_cap_reg_define(struct nd *tgt, int idx, uint8_t flags)
+{
+	struct ff_head          *h        = ff_data(tgt);
+	struct node_capture_reg *cap      = &tgt->n_cap;
+	struct ff_dir_entry     *de       = (struct ff_dir_entry *)(h + 1);
+	struct node_capture_reg  tmp      = {};
+	int                      crc_size = 0;
+
+	if (h->ff_var.ff_level == 0 &&
+	    ff_crctype_get(tgt) == M0_BCT_BTREE_ENC_RAW_HASH)
+		crc_size = CRC_VALUE_SIZE;
+
+	if ((flags & CR_HDR_MASK) == CR_HDR_FULL) {
+		tmp.ncr_hdr_beg = h;
+		tmp.ncr_hdr_end = h->ff_opaque == NULL ?
+				  tmp.ncr_hdr_beg + sizeof(*h) :
+				  (void *)(&h->ff_opaque) - 1;
```

Comment:
You can safely remove the check for h->ff_opaque == NULL  as it will never be NULL. So you can change the code to 
```suggestion
		tmp.ncr_hdr_end = tmp.ncr_hdr_beg + sizeof(*h);
```

Reply:
Done.